### ISSUE TITLE
Disable transformers evaluation when evaluator is provided to `SentenceTransformerTrainer`

### ISSUE THREAD

**Issue by emapco** [UKPLab/sentence-transformers#3459]
Is it possible to prevent `transformers` from evaluating the `eval_dataset` when an evaluator is provided to `SentenceTransformerTrainer`?

```py
stsb_eval_dataset = load_dataset("sentence-transformers/stsb", split="validation")
dev_evaluator = EmbeddingSimilarityEvaluator(
    sentences1=stsb_eval_dataset["sentence1"],
    sentences2=stsb_eval_dataset["sentence2"],
    scores=stsb_eval_dataset["score"],
    main_similarity=SimilarityFunction.COSINE,
    name="sts-dev",
    show_progress_bar=True,
)

trainer = SentenceTransformerTrainer(
    model=model,
    args=args,
    train_dataset=train_dataset,
    eval_dataset=eval_dataset,
    loss=train_loss,
    evaluator=dev_evaluator,
)
```

For example, when you run [examples/sentence_transformer/training/nli/training_nli_v3.py](https://github.com/UKPLab/sentence-transformers/blob/master/examples/sentence_transformer/training/nli/training_nli_v3.py), `transformers` runs its own evaluation before the `EmbeddingSimilarityEvaluator` evaluator is called. This results in the evaluation dataset(s) being run twice at every evaluation step.

In my use case, the initial `transformers` evaluation is unnecessary since the `metric_for_best_model` is computed by the `EmbeddingSimilarityEvaluator`.

**Comment by tomaarsen** [comment]
Hello!

Yes, this is definitely possible. You can avoid the "standard" transformers (loss) evaluation by not providing the `eval_dataset` to the `SentenceTransformerTrainer`, and only provider the trainer with the `evaluator`.

- Tom Aarsen

**Comment by emapco** [comment]
Thank you!
